### DISCUSSION TITLE
esp 32 not connecting to Mac M2

### DISCUSSION THREAD

**Discussion by damieaamri** [espressif/arduino-esp32#10987] [Question - Community Help]
### Board

ESP32 Dev Module

### Device Description

ESP32 Dev Module

### Hardware Configuration

ESP32

### Version

latest master (checkout manually)

### IDE Name

Arduino IDE

### Operating System

macOS Sonoma

### Flash frequency

-

### PSRAM enabled

yes

### Upload speed

-

### Description

Anyone can help i have downloaded all the driver that are available but still it doesnt detect any new port especially the UART.

only this is shown /dev/cu.Bluetooth-Incoming-Port when i prompt  ls /dev/cu.*

### Sketch

```cpp
none
```

### Debug Message

```plain
Sketch uses 267981 bytes (20%) of program storage space. Maximum is 1310720 bytes.
Global variables use 20184 bytes (6%) of dynamic memory, leaving 307496 bytes for local variables. Maximum is 327680 bytes.
esptool.py v4.6
Serial port /dev/cu.Bluetooth-Incoming-Port
Connecting......................................

A fatal error occurred: Failed to connect to ESP32: No serial data received.
For troubleshooting steps visit: https://docs.espressif.com/projects/esptool/en/latest/troubleshooting.html
Failed uploading: uploading error: exit status 2
```

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by Jason2866** [comment]
Driver for the USB serial chip installed?

**Comment by damieaamri** [comment]
i have tried downloading all of this:

- CH34xVCPDriver
- Mac_OSX_VCP_Driver
- SiLabsUSBDriverDIsk
- CH341SER_MAC

and the usb is connected via USB-hub.

it doesnt seem to find the port

**Comment by Jason2866** [comment]
driver can conflict with each other. The USB hub can do harm too.

**Comment by damieaamri** [comment]
> driver can conflict with each other. The USB hub can do harm too.

what should i do

**Comment by Jason2866** [comment]
Check which USB serial chip your board uses. Deinstall all drivers. This can be tricky.
Install only the driver which is the correct one for your dev board. Do not use a USB Hub.

**Comment by robertlipe** [comment]
Jason's on the right trail. There's a lot of guessing above at which driver to use, but first you should answer the fundamental question: what UART DOES your unnamed device actually use? That will shorten the search for the correct driver. Don't just install every driver you can find.

I use a stable of most the whole ESP32 family on MacOS without issue. (Well, I do know of ONE issue, but you're not having it, so I won't complicate things.)

Also, double check your cable actually passes data. Until System Settings shows a USB peripheral on the USB bus, the driver won't matter.

**Comment by Jason2866** [comment]
@robertlipe My M1 Mac does work with all dev boards I own. There are some since I am maintain the customized Tasmota Arduino Platformio Framework and do "some" testing ;-)

**Comment by robertlipe** [comment]
I recognize you, Jason. Your credentials are clearly solid. :-) I was trying to underscore your methodical approach to solving a problem instead of installing a dozen drivers when so far they hasn't even said what chip they have.

Let's not pull it into the discussion, but the 44-pin S3 devkit-c clones that use the WCH where the default driver *almost* works (well enough to do terminal stuff and start an upload, almost never well enough to finish an upload) until you use the new driver and the /dev/cu.wchWHATEVER nodes. 

That's the only excuse that comes to mind on any of my own Macs with any of the couple dozen dev boards I have. (I have no doubt your collection "wins". No need for a contest.)

**Comment by damieaamri** [comment]
where can i find the chip? is it on the esp32 board itself? im still new in developing IOT projects so im not sure how to fix the problem.

**Comment by Jason2866** [comment]
> where can i find the chip? is it on the esp32 board itself?

Yes, on the board itself. You can post an photo here.

This one ` /dev/cu.wchWHATEVER` tooks me a while to get fixed. The issue was to get the old driver fully deinstalled. After having ONLY the needed actual driver installed ALL boards works well to flash

**Comment by damieaamri** [comment]
<img width="311" alt="board" src="https://github.com/user-attachments/assets/0869a7bc-3654-4bc0-a58e-bc8a571da670">


> > where can i find the chip? is it on the esp32 board itself?
> 
> Yes, on the board itself. You can post an photo here.
> 
> This one ` /dev/cu.wchWHATEVER` tooks me a while to get fixed. The issue was to get the old driver fully deinstalled. After having ONLY the needed actual driver installed ALL boards works well to flash

**Comment by robertlipe** [comment]
Promise you're using this USB jack.
And the chip in question is probably this one in red. We don't need every number on it. Just the part that's probably largest and that returns ssomething about "serial chip" when you google it. Likely answers are CH340 or CH9102 or CP2102 or such...

<img width="311" alt="c" src="https://github.com/user-attachments/assets/1bca7db8-3076-48a5-a001-b05cd772e9bd">

**Comment by damieaamri** [comment]
its the CP2102

**Comment by robertlipe** [comment]
Now you're getting somewhere.

I've never heard of the site, but the recipe at
https://wiki.keyestudio.com/How_to_Install_the_Driver_of_CP2102_on_MAC_System
seems about right.

RJL

**Comment by damieaamri** [comment]
do i need to use the direct cable from usb-c  to the micro-usb or can i plug it on a USB Hub.

**Comment by lbernstone** [comment]
For troubleshooting, make a direct connection from the computer to the esp32.  Once you know it is working, you can test through the hub, using the other ports, etc.

**Comment by leezicai** [comment]
Someone said that the Thunderbolt interface doesn’t support direct connections. Have you tried using a USB hub to adapt it? That might work.

**Comment by damieaamri** [comment]
i tried both direct connection and USB HUb. but still nothing happen. i think the board is broken already.
